editor/editor_resource_picker.cpp

```diff
@@ -90,7 +90,8 @@ void EditorResourcePicker::_update_resource_preview(const String &p_path, const
 	if (p_preview.is_valid()) {
 		preview_rect->set_offset(SIDE_LEFT, assign_button->get_icon()->get_width() + assign_button->get_theme_stylebox(SNAME("normal"))->get_default_margin(SIDE_LEFT) + get_theme_constant(SNAME("hseparation"), SNAME("Button")));
 
-		if (Ref<GradientTexture1D>(edited_resource).is_valid()) {
+		// Ressource-specific stretching.
+		if (Ref<GradientTexture1D>(edited_resource).is_valid() || Ref<Gradient>(edited_resource).is_valid()) {
```

Comment:
It would be nice to make stretching be defined by EditorResourcePreviewGenerator instead of hardcoding the resources, so that plugins can add stretching previews for custom resources too. But that can be done in another PR.

Reply:
You're right, I thought of this too, but this PR is just for adding the preview. I actually plan to refactor the resource preview system (as there are also some other small issues) sometime in the following weeks, it's already on my to-do list :)